app/mac/scripts/codeSign.js

```diff
@@ -0,0 +1,25 @@
+require('dotenv').config();
+const { execSync } = require('child_process');
+const path = require('path');
+
+exports.default = async function codeSign(config) {
+  const teamID = process.env.APPLE_TEAM_ID;
+
+  if (!teamID) {
+    console.log('Mac codesign: No Apple Team ID found, skipping codesign');
+    return;
+  }
+
+  const entitlementsPath = path.resolve(path.join(__dirname, '..', 'entitlements.mac.plist'));
+
+  let exitCode = 0;
+  try {
+    execSync(
+      `codesign -s ${teamID} --deep --force --options runtime --entitlements ${entitlementsPath} ${config.app}`
```

Comment:
This is for building the app only and the code is not shipped in Headlamp. So I don't think there's a possibility of attack as it could be if the script was called during the execution of Headlamp.